src/osgEarth/MPScreenSpaceLayoutSG.cpp

```diff
@@ -482,14 +494,80 @@ struct /*internal*/ MPDeclutterSortSG : public osgUtil::RenderBin::SortCallback
                 updateOffsetForAutoLabelOnLine(box, vp, annoDrawable->_cull_anchorOnScreen, annoDrawable, camVPW, slidingOffset, to);
                 annoDrawable->_cull_anchorOnScreen += slidingOffset;
             }
-            
+
+            // Computes label location for the grid mora (visible part of the polygon)
+            if (annoDrawable->polygonVisible())
+            {
+                double maxAlt = annoDrawable->getTextPolygonAltitude();
+                double currentAltitude;
+                cam->getUserValue("altitude", currentAltitude);
```

Comment:
Not here !
It is camera related and is then const for the whole frame.
(should be moved where the eye, look,... are computed)